torch/utils/_sympy/interp.py

```diff
@@ -0,0 +1,92 @@
+"""
+This is a simple interpreter for Sympy expressions that dispatches to
+classes following the torch._inductor.virtualized calling convention.
+For directness, the interpreter takes the handler directly rather than
+consulting the TLS.  It does not use most of the methods on the full
+handler; only those with corresponding Sympy expressions.  To see an example
+of a full handler, see torch.utils._sympy.value_ranges.ValueRangeAnalysis.
+"""
+
+import functools
+from typing import Any, Dict, Union
+
+import sympy
+from sympy.logic.boolalg import BooleanAtom
+
+import torch
+
+
+SympyBoolean = sympy.logic.boolalg.Boolean
+
+
+# TODO: Dedupe this with SYMPY_INTERP
+
+
+@functools.lru_cache(None)
+def handlers():
+    from torch.fx.experimental.symbolic_shapes import FloorDiv, Pow, TrueDiv
+
+    HANDLERS = {
+        sympy.Or: "or_",
+        sympy.And: "and_",
+        sympy.Eq: "eq",
+        sympy.Ne: "ne",
+        sympy.Lt: "lt",
+        sympy.Gt: "gt",
+        sympy.Le: "le",
+        sympy.Ge: "ge",
+        sympy.Not: "not_",
+        TrueDiv: "truediv",
+        FloorDiv: "div",
+        sympy.Add: "add",
+        sympy.Mul: "mul",
+        Pow: "pow",
+        sympy.Pow: "pow",
+        sympy.Mod: "mod",
+        sympy.Abs: "abs",
+        sympy.log: "log",
+        sympy.exp: "exp",
+        sympy.floor: "floor",
+        sympy.ceiling: "ceil",
+        sympy.Min: "minimum",
+        sympy.Max: "maximum",
+    }
+    return HANDLERS
+
+
+ASSOCIATIVE_OPS = {"minimum", "maximum", "mul", "add", "and_", "or_"}
+
+
+def sympy_interp(
+    analysis, bindings: Dict[sympy.Symbol, Any], expr: Union[sympy.Expr, SympyBoolean]
+):
+    # Handle base cases
+    # TODO: not really sure if I'm passing the right dtype here
+    # TODO: wouldn't it be better to pass the sympy expression through
+    # sometimes?
```

Comment:
So when I was testing, I was running everything with sympy expressions to avoid disagreements between float and sympy symbolic. But because constant() only takes int/float, I have to int/float the sympy expression before calling it into constant (and then the reference analysis turns it back into sympy).